packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2729,8 +2729,12 @@ class InputDecoration {
   /// [errorText], and [counterText].
   ///
   /// By default the `contentPadding` reflects [isDense] and the type of the
-  /// [border]. If [isCollapsed] is true then `contentPadding` is
-  /// [EdgeInsets.zero].
+  /// [border]. If the [border] is [OutlineInputBorder] then `contentPadding` is
+  /// `EdgeInsets.LTRB(12, 24, 12, 16)`. If the [border] is [UnderlineInputBorder]
+  /// or [InputBorder.none] then `contentPadding` is `EdgeInsets.LTRB(0, 12, 0, 12)`.
+  /// If the [border] is [UnderlineInputBorder] or [InputBorder.none] and if
+  /// [filled] is true then `contentPadding` is `EdgeInsets.LTRB(12, 12, 12, 12)`.
+  /// If [isCollapsed] is true then `contentPadding` is [EdgeInsets.zero].
```

Comment:
I think that the existing wording could use some work as well. The documentation assumes that by default, padding is applied where `isDense` is true. I think we should just lay it out as the code says so that people aren't confused by the behavior here.

```
When `contentPadding` is not specified, its default value depend on the type 
of [border] used, the value of [isDense], and the value of [isCollapsed].

When [isCollapsed] is true, the content padding defaults to [EdgeInsets.zero]. 

When [border.isOutline] is false and [filled] is true, then content padding will 
be set to `EdgeInsets.fromLTRB(12, 8, 12, 8)` when [isDense] is true and 
`EdgeInsets.fromLTRB(12, 12, 12, 12)` when [isDense] is false. When 
[border.isOutline] is false and [filled] is false, then content padding will 
be the same as above, but without the left and right padding.

When [border.isOutline] is true, then content padding will be set 
to `EdgeInsets.fromLTRB(12, 20, 12, 12)` when [isDense] is true
and `EdgeInsets.fromLTRB(12, 24, 12, 16)` when [isDense] is false.
```

Let me know what you think, but definitely give your suggestions and only make modifications as you see fit :)

Reply:
I think you are right, the first statement is ambiguous. We can list out all the possible default values of `contentPaddding`